### PR TITLE
Parallel validation for Xcode installation

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
# Description
In scope of this pull request we add logic to parallelise validation for Xcode during its installation.

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/xcode.ps1 [hunk 1]
```diff
@@ -14,20 +14,28 @@ $os = Get-OSVersion
 $xcodeVersions = Get-ToolsetValue "xcode.versions"
 $defaultXcode = Get-ToolsetValue "xcode.default"
 [Array]::Reverse($xcodeVersions)
+$threadCount = 8
```
Comment (maxim-lobanov) [inline images/macos/provision/core/xcode.ps1:17]: There is a function to return the number of processors, may be it is better to use it?
image

File: images/macos/helpers/Xcode.Installer.psm1 [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Import-Module "$PSScriptRoot/Xcode.Helpers.psm1"
+Import-Module "~/image-generation/helpers/Xcode.Helpers.psm1"
```
Comment (maxim-lobanov) [inline images/macos/helpers/Xcode.Installer.psm1:1]: Let's revert this line if we don't need it

File: images/macos/provision/core/xcode.ps1 (hunk 1 above)
Comment (al-cheb) [reply images/macos/provision/core/xcode.ps1:17]: nitpick: `[Environment]::ProcessorCount` as a variant

Comment (maxim-lobanov) [reply images/macos/provision/core/xcode.ps1:17]: Yep, `[Environment]::ProcessorCount` is even better from readability side

**Pull request closed without merging**